lib/internal/encoding.js

```diff
@@ -337,7 +341,12 @@ class TextEncoder {
 
   encode(input = '') {
     validateEncoder(this);
-    return encodeUtf8String(`${input}`);
+    const utf8Length = getUtf8Length(input);
+    const array = new Uint8Array(utf8Length);
```

Comment:
I can assume input.length as the `new Uint8Array` length, and fallback to slow path, if the it requires more size. But it would ideally slow the `slow path`, and I'm not happy with it.